examples/vision/yolov8.py

```diff
@@ -0,0 +1,628 @@
+KerasCV.
+
+"""
+
+yolo = keras_cv.models.YOLOV8Detector(
+    num_classes=len(class_mapping),
+    bounding_box_format="xyxy",
+    backbone=backbone,
+    fpn_depth=1,
+)
+
+"""
+## Compile the Model
+"""
+
+"""
+Loss used for YOLOV8
+
+
+1. Classification Loss: This loss function calculates the discrepancy between anticipated
+class probabilities and actual class probabilities. In this instance,
+`binary_crossentropy`, a prominent solution for binary classification issues, is
+Utilized. We Utilized binary crossentropy since each thing that is identified is either
+classed as belonging to or not belonging to a certain object class (such as a person, a
+car, etc.).
+
+2. Box Loss:  `box_loss` is the loss function used to measure the difference between the
+predicted bounding boxes and the ground truth. In this case, the Intersection over Union
+(IoU) metric is used, which measures the overlap between predicted and ground truth
+bounding boxes. Together, these loss functions help optimize the model for object
+detection by minimizing the difference between the predicted and ground truth class
+probabilities and bounding boxes.
+
+
+"""
+
+optimizer = tf.keras.optimizers.Adam(
+    learning_rate=LEARNING_RATE,
+    global_clipnorm=GLOBAL_CLIPNORM,
+)
+
+yolo.compile(
+    optimizer=optimizer, classification_loss="binary_crossentropy", box_loss="iou"
+)
+
+"""
+## COCO Metric Callback
+
+We will be using `BoxCOCOMetrics` from KerasCV to evaluate the model and calculate the
+Map(Mean Average Precision) score, Recall and Precision.
+"""
+
+
+class EvaluateCOCOMetricsCallback(keras.callbacks.Callback):
+    def __init__(self, data):
+        super().__init__()
+        self.data = data
+        self.metrics = keras_cv.metrics.BoxCOCOMetrics(
+            bounding_box_format="xyxy",
+            evaluate_freq=1e9,
+        )
+
+    def on_epoch_end(self, epoch, logs):
+        self.metrics.reset_state()
+        for batch in self.data:
+            images, y_true = batch[0], batch[1]
+            y_pred = self.model.predict(images, verbose=0)
+            self.metrics.update_state(y_true, y_pred)
+
+        metrics = self.metrics.result(force=True)
+        logs.update(metrics)
+        return logs
+
+
+"""
+## Train the Model
+"""
+
+yolo.fit(
+    train_ds,
+    validation_data=val_ds,
+    epochs=EPOCH,
+    callbacks=[EvaluateCOCOMetricsCallback(val_ds)],
+)
+
+"""
+## Visualize Predictions
+"""
+
+
+def visualize_detections(model, dataset, bounding_box_format):
+    images, y_true = next(iter(dataset.take(1)))
+    y_pred = model.predict(images)
+    y_pred = bounding_box.to_ragged(y_pred)
+    visualization.plot_bounding_box_gallery(
+        images,
+        value_range=(0, 255),
+        bounding_box_format=bounding_box_format,
+        y_true=y_true,
+        y_pred=y_pred,
+        scale=4,
+        rows=2,
+        cols=2,
+        show=True,
+        font_scale=0.7,
+        class_mapping=class_mapping,
+    )
+
```

Comment:
@ianstenbit 

Does yolo-v8 support changing NMS afterward? I tried onec, didn't work with yolo-v8 but worked with retinanet. 

```python
keras_cv.layers.MultiClassNonMaxSuppression(
    bounding_box_format="xywh",
    from_logits=True,
    iou_threshold=0.5,
    confidence_threshold=0.75,
)
``

Reply:
Ahh good catch -- I'll send a fix for this now.

Reply:
Sent [a KerasCV PR](https://github.com/keras-team/keras-cv/pull/1841) to fix this